docs: Remove UI docs

tayloramurphy opened the pull request:
Meltano is focusing on improving the CLI as the primary way for users to control and interact with their Meltano project. Many of the newer features in Meltano (`run`, environments, mappers, etc) and MeltanoHub (api) do not work with the Meltano UI and create a bad user experience. 

In the future we will prioritize having a UI, but it will be rebuilt based on our current and future featureset as a complement to the CLI. We invite discussion from the community on the future UI in [this discussion](https://github.com/meltano/meltano/discussions/6957).

This PR removes the documentation around Meltano UI.

Closes https://github.com/meltano/meltano/issues/6924

Commit:
Remove UI docs

Commit:
Remove more UI docs

Commit:
Add UI notice in troubleshooting

Commit:
Remove more UI docs

Commit:
Remove email docs

Commit:
Merge branch 'main' into remove-ui-docs

Commit:
Cleanup plugin contribution docs

aaronsteers commented on docs/src/_reference/command-line-interface.md:
```diff
@@ -1341,53 +1341,6 @@ meltano test <plugin1>:<test-name1> <plugin2>:<test-name2>
 
 The `test` command can accept the `--environment` flag to target a specific [Meltano Environment](https://docs.meltano.com/concepts/environments). The [`default_environment` setting](https://docs.meltano.com/concepts/environments#default-environments) in your `meltano.yml` file will be applied if `--environment` is not provided explicitly.
 
-## `ui`
-
-- `meltano ui`: Start the Meltano UI.
-
```
@tayloramurphy , what do you think of keeping this small reference, but with a deprecation note and a pointer to the troubleshooting section?

tayloramurphy replied:
I'm cool with that!

aaronsteers reviewed (commented):
@tayloramurphy - On the whole, I think the vast majority of these changes are great and should definitely be merged. There were places where we were unnecessarily promoting the web UI when the better course for the user is just to use the CLI (or yaml, but that's another topic.) 

I do feel like there are certain parts that I'd like to keep around _somewhere_ as reference - for our internal reference if nothing else.

Also - we know there are some number of users interacting with HTTP endpoints that are the web API. And we (soon!) will start working on a REST API interface which might (or might not) build from this as a foundation.

Perhaps we could redirect all 'UI' references to troubleshooting, while putting any helpful reference info for us or for our users into something like a `_reference/server.md` that is omitted from the left-hand index but linked to from `troubleshooting.md`.

I'm also open to other ideas and suggestions as well. Thanks!

sbalnojan commented on docs/src/_concepts/project.md:
```diff
@@ -354,15 +354,6 @@ Meltano's CLI utilizes the following tables:
 
 - `runs` table: One row for each [`meltano elt`](/reference/command-line-interface#elt) or [`meltano run`](/reference/command-line-interface#run) pipeline run, holding started/ended timestamps and [incremental replication state](/guide/integration#incremental-replication-state).
 - `plugin_settings` table: [Plugin configuration](/guide/configuration#configuration-layers) set using [`meltano config <plugin> set`](/reference/command-line-interface#config) or [the UI](/reference/ui) when the project is [deployed as read-only](/reference/settings#project-readonly).
-- `user` table: Users for [Meltano UI](/reference/ui) created using [`meltano user add`](/reference/command-line-interface#user).
-
-The remaining tables in the database are used exclusively by Meltano UI, mostly for authentication and authorization purposes:
-
-- `role`
-- `role_permissions`
-- `oauth`
-- `embed_tokens`
-- `subscriptions`
```
@aaronsteers Why not put them into an issue for removing the rest of the UI functionality?

tayloramurphy replied:
@aaronsteers my assumption is that having them in version control should be enough. I've made https://github.com/meltano/meltano/issues/6959 to track to the work to remove the code bits and cross-linked the PR.

aaronsteers replied:
@sbalnojan and @tayloramurphy - I think the crux of this for me is _how long_ until V3 - which is when I thought we'd _remove_ the code and capability.

I don't see that as coming soon personally, with everything else we have on our plates right now. If it is possibly 6 months out, the cost for us and users in the meanwhile is not insignificant. We could hide this behind a feature flag between now and V3 (forcing a bit of extra effort to get the UI to run), but still we have a period of support we need to be accountable for between now and then.

So, my preference would be:

- Yes, we can _remove_ any and all 'how to guides', 'tutorials', etc. These are things that help new users onboard to the features and discover them for the first time.
- Let's preserve all _reference docs_ that we (Meltano team) and existing users of the features will need in order to support currently-onboarded implementations.

tayloramurphy commented on docs/src/_reference/command-line-interface.md:
```diff
@@ -1341,53 +1341,6 @@ meltano test <plugin1>:<test-name1> <plugin2>:<test-name2>
 
 The `test` command can accept the `--environment` flag to target a specific [Meltano Environment](https://docs.meltano.com/concepts/environments). The [`default_environment` setting](https://docs.meltano.com/concepts/environments#default-environments) in your `meltano.yml` file will be applied if `--environment` is not provided explicitly.
 
```
```suggestion

## `ui` (deprecated)

The Metano UI is now deprecated. For more information see our [troubleshooting page](/guide/troubleshooting#meltano-ui).
```

Commit:
Update docs/src/_reference/command-line-interface.md

Commit:
Update docs/src/_guide/troubleshooting.md

Co-authored-by: Aaron ("AJ") Steers <aj@meltano.com>

tayloramurphy commented:
> And we (soon!) will start working on a REST API interface which might (or might not) build from this as a foundation.

> Perhaps we could redirect all 'UI' references to troubleshooting, while putting any helpful reference info for us or for our users into something like a _reference/server.md that is omitted from the left-hand index but linked to from troubleshooting.md.

@aaronsteers I don't think I can do the redirects at the header level - I think it's only for the full pages. I can test that though. 

Until we start working more on the API my sense is that it's okay to remove the pages as we'll always have them in version control.

Commit:
test anchor redirect

aaronsteers commented:
> > And we (soon!) will start working on a REST API interface which might (or might not) build from this as a foundation.
> 
> > Perhaps we could redirect all 'UI' references to troubleshooting, while putting any helpful reference info for us or for our users into something like a _reference/server.md that is omitted from the left-hand index but linked to from troubleshooting.md.
> 
> @aaronsteers I don't think I can do the redirects at the header level - I think it's only for the full pages. I can test that though.
> 
> Until we start working more on the API my sense is that it's okay to remove the pages as we'll always have them in version control.

As I note in my comment, I don't want to remove 'reference' materials because it just unnecessarily makes things harder for us to manage. I am fine removing everything else, and I'm totally fine with us removing all top-level nav that would point to the reference. (And of course, we'd have a banner at the top of the reference page that says all the below is being deprecated, etc.)

tayloramurphy commented on docs/src/_guide/troubleshooting.md:
```diff
@@ -2,6 +2,9 @@
 layout: doc
+redirect_from:
+ - /reference/ui
+ - /guide/orchestration#meltano-ui
```
```suggestion
```
confirmed that you can't redirect from an anchor

Commit:
Update docs/src/_guide/troubleshooting.md

Commit:
merge conflict

tayloramurphy commented:
@aaronsteers I've updated and moved most of the technical content to https://deploy-preview-6955--meltano.netlify.app/reference/ui Let me know what you think.

aaronsteers committed:
Merge branch 'main' into remove-ui-docs

aaronsteers commented on docs/src/_reference/ui.md:
```diff
@@ -2,11 +2,14 @@
 title: Meltano UI
 description: Learn how to manage and monitor your plugins and pipelines using Meltano UI
```
```suggestion
```

```suggestion
title: Meltano UI and REST API Server (Deprecated)
description: Reference page for the deprecated Meltano UI and REST API Server configuration settings.
```

aaronsteers commented on docs/src/_reference/ui.md:
```diff
@@ -2,11 +2,14 @@
-  - /guide/ui/
-weight: 3
+hidden: true
+toc: false
```
Okay to enable TOC on this page?

```suggestion
toc: true
```

aaronsteers commented on docs/src/_contribute/ui.md:
```diff
@@ -1,126 +0,0 @@
-<div class="notification is-warning">
-  <p><strong>If you're developing for the _Embed_ app (embeddable <code>iframe</code> for reports) you can toggle <code>MELTANO_EMBED</code>:</strong></p>
-
-<pre>
-# Develop for the embed app
-export MELTANO_EMBED=1
-
-# Develop for the main app (this is the default)
-export MELTANO_EMBED=0
-
-# Start local development environment
-yarn serve
-</pre>
-</div>
```
Difficult to do a thorough comparison but I didn't see this contributing content (`yarn serve`, etc.) and could be valuable to preserve in reference page:

- https://deploy-preview-6955--meltano.netlify.app/reference/ui

aaronsteers reviewed (commented):
A couple minor changes requested. Thanks for the additional changes here to preserve an internal reference for UI and REST API aspects.

I think there is some content in `docs/src/_contribute/ui.md` which still might be worth preserving but I haven't been able to do a thorough enough review of that page to know how much we would preserve. I do think some of it is important, though, such as how to use `yarn` to serve the repo from the dev project.

Happy to take this on if you'd like. I just wanted to make sure I got back to you back on this PR before you were OOO.

aaronsteers reviewed (changes requested):
See above ☝️

Commit:
Update docs/src/_reference/ui.md

Co-authored-by: Aaron ("AJ") Steers <aj@meltano.com>

Commit:
Update docs/src/_reference/ui.md

Co-authored-by: Aaron ("AJ") Steers <aj@meltano.com>

Commit:
Update docs/src/_reference/plugin-definition-syntax.md

Co-authored-by: Aaron ("AJ") Steers <aj@meltano.com>

Commit:
Add contributing docs to reference

aaronsteers reviewed (approved):
@tayloramurphy - Looks good! Thanks for tackling this!

Commit:
Merge branch 'main' into remove-ui-docs
